homeassistant/components/huawei_lte/__init__.py

```diff
@@ -150,40 +150,53 @@ def device_connections(self) -> Set[Tuple[str, str]]:
         """Get router connections for device registry."""
         return {(dr.CONNECTION_NETWORK_MAC, self.mac)} if self.mac else set()
 
+    def _get_data(self, key: str, func: Callable[[None], Any]) -> None:
+        if not self.subscriptions.get(key):
+            return
+        _LOGGER.debug("Getting %s for subscribers %s", key, self.subscriptions[key])
+        try:
+            self.data[key] = func()
+        except ResponseErrorNotSupportedException:
+            _LOGGER.info(
+                "%s not supported by device, excluding from future updates", key
+            )
+            self.subscriptions.pop(key)
+        except ResponseErrorLoginRequiredException:
+            if isinstance(self.connection, AuthorizedConnection):
```

Comment:
If you do it the other way around, it does seem to get slightly less messy (`if not isinstance(…)`). Personal preference I guess

Reply:
Maybe, dunno, but doing it the other way around requires one more `else` block. No strong opinions but I think I'll just leave it as is now.